### ISSUE TITLE
Issues with cross build `Conflicting cross-version suffixes in: org.scala-lang.modules:scala-xml`

### ISSUE THREAD

**Issue by stanislav-chetvertkov** [scapegoat-scala/sbt-scapegoat#70]
I'm having issues with cross build `crossScalaVersions := Seq("2.11.11", "2.12.4"),`

It gives `Conflicting cross-version suffixes in: org.scala-lang.modules:scala-xml`

I tried excluding it from `plugins.sbt` like
```
addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.0.7"
  exclude("org.scala-lang.modules", "scala-xml"))
```

Also tried to add `exclude("org.scala-lang.modules", "scala-xml")` from every library that has scala-xml in it and that also didn't help.

Here is a simplified version of how build.sbt looks like:

```
scapegoatVersion in ThisBuild := "1.3.4"

lazy val commonSettings = Seq(
  dependencyOverrides += "org.scala-lang" % "scala-compiler" % scalaVersion.value,
  scalacOptions := Seq("-target:jvm-1.8", "-encoding", "utf8", "-feature", "-unchecked", "-deprecation", "-explaintypes", "-Xfatal-warnings"),
  organization := "com.example",
  scalaVersion := "2.12.4",
)

lazy val protocol = (project in file("protocol"))
  .settings(commonSettings)
  .settings(
    crossScalaVersions := Seq("2.11.11", "2.12.4"),
    name := "protocol",
    moduleName := "protocol",
    libraryDependencies ++= Seq(
      "io.circe" %% "circe-java8" % circeVersion,
      "io.circe" %% "circe-generic" % circeVersion,
      "io.circe" %% "circe-parser" % circeVersion,
      "io.circe" %% "circe-derivation" % "0.9.0-M1",
      "io.circe" %% "circe-generic-extras" % "0.9.1",
      "org.scalatest" %% "scalatest" % scalaTestVersion % Test
    )
  )

lazy val client = (project in file("client"))
  .settings(commonSettings)
  .settings(
    crossScalaVersions := Seq("2.11.11", "2.12.4"),
    name := "client",
    moduleName := "client",
    libraryDependencies += "net.databinder.dispatch" %% "dispatch-core" % "0.12.0"  exclude("org.scala-lang.modules", "scala-xml")
    }.value,
   (compile in Compile) := (compile in Compile).dependsOn(scapegoat).value
  ).dependsOn(protocol)

lazy val root = (project in file("."))
  .settings(commonSettings)
  .settings(
    name := "service",
    moduleName := "service"
  ).aggregate(protocol, client)
```

running just `sbt assembly` with no crossbuild works fine.
Also removing `(compile in Compile) := (compile in Compile).dependsOn(scapegoat).value` does not affect the cross build - it still fails, only removing the plugin from `plugins.sbt` makes it work.

Will appreciate any feedback, thanks

**Comment by sahalsaad** [comment]
Try add  `scalaBinaryVersion in ThisBuild := "2.12"` to your build.sbt. It fix the problem for me.

**Comment by diversit** [comment]
Running into the same issue. In a cross-build the scapegoat dependency is not correctly changing to a different scala version. When building for Scala 2.12 all is ok. Then when building for 2.11, still the scapegoat_2.12 dependency is used.

Using `show dependencyList` on the project shows that all other dependencies have changed to '_2.11', but scapegoat is still on 'com.sksamuel.scapegoat:scalac-scapegoat-plugin_2.12:1.3.7'.

Since scapegoat pulls in the scala-xml_2.12 lib and something else pulls in scala-xml_2.11 this conflict occurs.

Suggested fix above did not work for me. So meanwhile, no scapegoat checks for me :-(

**Comment by drissnejjar** [comment]
Hello, I have the same problem. Any feedback for this issue ?

Thanks

**Comment by Johnnei** [comment]
Cross build support was improved with #155 (released in 1.2.0).
